spock-core/src/main/java/org/spockframework/runtime/ParameterizedFeatureNode.java

```diff
@@ -30,30 +38,106 @@ public SpockExecutionContext execute(SpockExecutionContext context, DynamicTestE
     if (testExecutor.getExecutionCount() < 1) {
      throw new SpockExecutionException("Data provider has no data");
     }
+
+    // do not try to aggregate iteration results if they are reported individually
+    if (featureInfo.isReportIterations()) {
+      return context;
+    }
+
+    if (testExecutor.results.containsKey(FAILED)) {
```

Comment:
The method `execute()` is quite big after your addition. What do you think about extracting your fragment to something like:
```
...
if (testExecutor.results.containsKey(FAILED)) {
    handleFailed(testExecutor.results)
} else if (!testExecutor.results.containsKey(SUCCESSFUL)) {
    handleAborted(testExecutor.results)
}
return context;
...
```
?

(I'm looking at that in a browser, so I might miss some obvious thing that could make it non practical)

Reply:
You are right, I extracted those blocks into methods.